api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -100,7 +102,11 @@ private void validateHelper(Obs obs, Errors errors, List<Obs> ancestors, boolean
 		if (obs.getObsDatetime() == null) {
 			errors.rejectValue("obsDatetime", "error.null");
 		}
-		
+		if (obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
```

Comment:
```suggestion
		if (obs.getPerson() != null && obs.getPerson().getIsPatient() && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
```

Reply:
Ideally, we shouldn't *need* to fix the other tests; at least this will stop the failure when the obs relates to a person who is not a patient.